core/blocksigner/enclave.go

```diff
@@ -0,0 +1,57 @@
+package blocksigner
+
+import (
+	"context"
+	"time"
+
+	"chain/core/rpc"
+	"chain/crypto/ed25519"
+	"chain/encoding/json"
+	"chain/protocol/bc/legacy"
+)
+
+// enclaveTimeout is the time to wait for an HSM's response to a
+// sign-block request before moving on to the next HSM. Because
+// blocks are expected to be generated every second, one second
+// retireving a signature would be a very high latency.
+const enclaveTimeout = time.Second
```

Comment:
Maybe instead we should request a signature from all enclaves in parallel, and let the caller of Sign impose whatever overall timeout they want. That would reduce latency when there's a failure, and when one of the enclaves is just slow.